pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1367,6 +1367,43 @@ def func(ser):
     tm.assert_frame_equal(res, expected)
 
 
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "int64",
+        "uint64",
+        "int32",
+        "uint32",
+        "float64",
+        "float32",
+        "Int64",
+        "UInt64",
+        "Int32",
+        "UInt32",
+        "Float64",
+        "Float32",
+    ],
+)
+def test_groupby_agg_precision(dtype):
+    # GH33234
+    df = DataFrame(
+        {
+            "key1": ["a"],
+            "key2": ["b"],
+            "key3": [
+                1583715738627261039
+                if dtype not in ["int32", "uint32", "Int32", "UInt32"]
+                else 4294967295
+                if dtype in ["uint32", "UInt32"]
+                else 2147483647
+            ],
+        }
+    ).astype({"key3": dtype})
```

Comment:
Can you define all those objects explicitly? e.g. no astype

Reply:
Ok. Could I  drop the astype part and keep just DataFrame creation expression?

Reply:
I think I defined all those objects explicitly in the updated commit.